utilities/ipplan-import.py

```diff
@@ -0,0 +1,181 @@
+"""
+NIPAP import script for IPplan data.
+
+This script uses the standard export files base.txt and ipaddr.txt from Ipplan.
+"""
+import configparser
+import os
+# use local pynipap, useful if we are developing
+import sys
+sys.path.append('../pynipap')
+import pynipap
+import csv
+import ipaddress
+
+
+def add_ip_to_net(networks, host):
+    """
+    Add hosts from IPPlan to networks object.
+    """
+    for network in networks:
+        if host['ipaddr'] in network['network']:
+            network['hosts'].append(host)
+            return
+
+
+def get_networks(base_file, ipaddr_file):
+    """
+    Gather network and host information from IPPlan export files.
+    """
+    networks = []
+
+    base = open(base_file, 'r')
+
+    csv_reader = csv.reader(base.readlines(), delimiter='\t')
+    for row in csv_reader:
+
+        # Rows with less than 3 entries is most likely a newline.
+        if len(row) < 3:
+            continue
+
+        network = {
+            'network': ipaddress.ip_network("{}/{}".format(row[0], row[2])),
+            'comment': row[1],
+            'hosts': []
+        }
+
+        networks.append(network)
+
+    base.close()
+
+    ipaddr = open(ipaddr_file, 'r')
+
+    csv_reader = csv.reader(ipaddr.readlines(), delimiter='\t')
+    for row in csv_reader:
+
+        host = {
+            'ipaddr': ipaddress.ip_address(row[0]),
+            'user': row[1],
+            'location': row[2],
+            'description': row[3],
+            'fqdn': row[4],
+            'phone': row[5],
+            'mac': row[6]
+        }
+
+        if len(row) > 7:
+            host['additional'] = " ".join(row[7:])
+
+        add_ip_to_net(networks, host)
+
+    ipaddr.close()
+
+    return networks
+
+def new_prefix():
+    """
+    Create new prefix object with general settings.
+    """
+    p = pynipap.Prefix()
+    p.authorative_source = "nipap"
+    return p
+
+
+def add_prefix(network):
+    """
+    Put your network information in the prefix object.
+    """
+    p = new_prefix()
+    p.prefix = str(network['network'])
+    p.type = "assignment"
+    p.description = network['comment']
+    p.tags = ['ipplan-import']
+    return p
+
+def add_host(host):
+    """
+    Put your host information in the prefix object.
+    """
+    p = new_prefix()
+    p.prefix = str(host['ipaddr'])
+    p.type = "host"
+    p.description = host['description']
+    p.node = host['fqdn']
+    p.comment = ""
+
+    # Use remaining data from ipplan to populate comment field.
```

Comment:
i.e. create an AVP for each attribute, like phone, mac and so forth.. don't cram it all together.